### PR TITLE
S-X-PM

### PR DESCRIPTION
**Pull request by MHoude2**
Changed Version number to 0.1.1

Changed original code to output JSA rather than abs(JSA) and modified the previous examples accordingly.

New function, SXPM_prop, generates the Heisenberg propagator when including self and cross-phase modulations. 

New file, example_sxpm.ipynb, shows how to use the new function briefly.

New tests to verify that the new function works properly.

### CODE REVIEW

File: NeedALight/tests/test_propagator.py [hunk 1]
```diff
@@ -74,9 +69,204 @@ def test_JSA_lowgain(N, Np):
...
+    )
+    pumpz = expm(
+        1j
+        * (spm * (w[1] - w[0]) / vp**2)
+        * (z2 + l / 2)
+        * density(-w + w[:, np.newaxis])
+    ) @ pump(w)
+
+    assert np.allclose(beta, pumpz)
+
```
Comment (nquesada) [inline NeedALight/tests/test_propagator.py]: I would have expected a Fourier transform here, no?

Comment (MHoude2) [reply NeedALight/tests/test_propagator.py]: It's in there but manually. I could change it to include your Fourier.py function but then I'd have to include that somewhere just for the test.

Comment (nquesada) [reply NeedALight/tests/test_propagator.py]: Could you add a comment in the relevant line saying you are doing the FT "by hand"?

After that, I am happy to approve.

Comment (MHoude2) [reply NeedALight/tests/test_propagator.py]: Should have been added in line 118

**Review by nquesada — APPROVED** [review]

**Pull request merged**